selfhost/parser.jakt

```diff
@@ -2166,14 +2160,30 @@ struct Parser {
         let start_span = .current().span()
         .index++
 
-        guard .current() is Identifier(name) else {
-            .error("Expected iterator name", .current().span())
-            return ParsedStatement::Garbage(merge_spans(start_span, .current().span()))
+        mut iterator_name = ""
+        mut destructured_var_decls: [ParsedVarDecl] = []
+        
+        match .current() {
+            Identifier(name) => {
+                iterator_name = name
+                .index++
+            }
+            LParen => {
+                destructured_var_decls = .parse_destructuring_assignment(is_mutable: false)
+                mut tuple_var_name = ""
+                for var in destructured_var_decls.iterator() {
+                    tuple_var_name += var.name
+                    tuple_var_name += "__"
+                }
+                iterator_name = tuple_var_name
+            }
+            else => {
+                .error("Expected iterator name", .current().span())
```

Comment:
Iterator name or destructuring pattern

Reply:
I updated the error message to better represent the need